R/spark_compile.R

```diff
@@ -367,3 +373,17 @@ get_scalac_version <- function(scalac = Sys.which("scalac")) {
   splat <- strsplit(version_string, "\\s+", perl = TRUE)[[1]]
   splat[[4]]
 }
+
+make_version_filter <- function(version_upper) {
```

Comment:
Since this is a function that returns a function, to be safe, you'll need to call `force(version_upper)` before the returned function definition.